src/NLog/Config/LoggingConfiguration.cs

```diff
@@ -361,10 +369,12 @@ internal void Dump()
         internal void FlushAllTargets(AsyncContinuation asyncContinuation)
         {
             var uniqueTargets = new List<Target>();
-            foreach (var rule in this.LoggingRules)
+            for (int i = 0; i < this.LoggingRules.Count; i++)
```

Comment:
I will fix it. (assigned myself also) 

Funny,  do your see an conflict? A here it says there are no conflicts in this PR.